vassal-app/src/main/java/VASSAL/configure/ImageSelector.java

```diff
@@ -0,0 +1,190 @@
+import VASSAL.tools.filechooser.FileChooser;
+import VASSAL.tools.filechooser.ImageFileFilter;
+import VASSAL.tools.image.ImageUtils;
+import VASSAL.tools.imageop.Op;
+import VASSAL.tools.imageop.OpIcon;
+import VASSAL.tools.imageop.ScaleOpBitmapImpl;
+
+import java.awt.Component;
+import java.awt.Dimension;
+import java.awt.event.ItemEvent;
+import java.awt.event.ItemListener;
+
+import javax.swing.BorderFactory;
+import javax.swing.DefaultComboBoxModel;
+import javax.swing.Icon;
+import javax.swing.ImageIcon;
+import javax.swing.JButton;
+import javax.swing.JComboBox;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+
+import net.miginfocom.swing.MigLayout;
+
+import org.apache.commons.lang3.ArrayUtils;
+
+/**
+ * Allows a user to select from the images currently available in a module, or
+ * to open a File Dialog to import a new image.
+ *
+ * Designed to be a drop-in replacement for {@link VASSAL.counters.ImagePicker},
+ * except implemented as a proper Configurer.
+ *
+ * The value stored is the name of the image file (no pathname).
+ */
+public class ImageSelector extends Configurer implements ItemListener {
+
+  private static final String NO_IMAGE = "(" + Resources.getString("Editor.ImagePicker.no_image") + ")";
+
+  private static final int DEFAULT_SIZE = 64;
+
+  private JPanel controls;
+  private JComboBox<String> select;
+  private Icon noImage;
+  private JLabel imageViewer;
+  private final JLabel imageScale = new JLabel();
+  private final OpIcon icon = new OpIcon();
+  private final int maxWidth;
+  private final int maxHeight;
+
+  public ImageSelector(String key, String name, String val, int maxWidth, int maxheight) {
+    super(key, name, val);
+    this.maxWidth = maxWidth;
+    this.maxHeight = maxheight;
+    initControls();
+    setValue(val);
+  }
+
+  public ImageSelector(String key, String name, String val) {
+    this(key, name, val, -1, -1);
+  }
+
+  public ImageSelector(String key, String name) {
+    this(key, name, null);
+  }
+
+  public ImageSelector(String val, int maxWidth, int maxheight) {
+    this(null, "", val, maxWidth, maxheight);
+  }
+
+  public ImageSelector(String val) {
+    this(null, "", val);
+  }
+
+
+  @Override
+  public String getValueString() {
+    return (String) value;
+  }
+
+  @Override
+  public void setValue(String s) {
+
+    if (s == null || s.isBlank() || s.equals(NO_IMAGE)) {
+      imageViewer.setIcon(getNoImageIcon());
+      imageViewer.setPreferredSize(new Dimension(DEFAULT_SIZE, DEFAULT_SIZE));
+    }
+    else {
+      icon.setOp(Op.load(s));
+
+      // Is the image too large?
+      if (maxWidth > 0 && (icon.getIconWidth() > maxWidth || icon.getIconHeight() > maxHeight)) {
+        final double xRatio = (double) maxWidth / icon.getIconWidth();
+        final double yRatio = (double) maxHeight / icon.getIconHeight();
+        final double newScale = Math.min(xRatio, yRatio);
+        icon.setOp(new ScaleOpBitmapImpl(Op.load(s), newScale));
```

Comment:
This will fail if the image is SVG. Use `Op.scale()` instead.

Reply:
Thanks, fixed.